src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -826,13 +844,19 @@ void Instance::HandleQueryIdentity(HandlerContext & ctx, const Commands::QueryId
 void Instance::OnResult(Status commissioningError, CharSpan debugText, int32_t interfaceStatus)
 {
     auto commandHandleRef = std::move(mAsyncCommandHandle);
-    auto commandHandle    = commandHandleRef.Get();
+
+    // In Non-concurrent mode the commandHandle will be null here, the ConnectNetworkResponse
+    // has already been sent and the BLE will have been stopped, however the other functionality
+    // is still required
+#if CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION
+    auto commandHandle = commandHandleRef.Get();
```

Comment:
What happens to the lifecycle of the commandHandleRef in this case?

Reply:
The commandHandleRef is completed when the NetworkConnectResponse(Success) is sent in SendNonConcurrentConnectNetworkResponse() prior to getting here.